File: backend/src/main/java/edu/cit/mabini/meditrack/appointment/AppointmentDto.java
package edu.cit.mabini.meditrack.appointment;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AppointmentDto {
    private Long id;

    @NotNull(message = "patient is required")
    private Long patientId;

    private String patientName;
    private String patientNumber;

    @NotNull(message = "doctor is required")
    private Long doctorId;

    private String appointmentNumber;

    @NotBlank(message = "appointmentType is required")
    private String appointmentType;

    private Integer queueNumber;

    @Positive(message = "durationMinutes must be greater than zero")
    private Integer durationMinutes;

    private Integer priority;
    private String notes;

    @NotNull(message = "appointmentDate is required")
    private LocalDate appointmentDate;

    @NotNull(message = "startTime is required")
    private LocalTime appointmentTime;

    @NotNull(message = "endTime is required")
    private LocalTime endTime;

    private Appointment.AppointmentStatus status;

    @NotBlank(message = "reason is required")
    private String remarks;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: mobile/app/src/main/java/edu/cit/mabini/meditrack/feature/appointments/data/model/AppointmentDto.kt
package edu.cit.mabini.meditrack.feature.appointments.data.model

data class AppointmentDto(
    val id: Long? = null,
    val patientId: Long = 0,
    val patientName: String? = null,
    val patientNumber: String? = null,
    val doctorId: Long = 0,
    val appointmentNumber: String? = null,
    val appointmentType: String = "",
    val queueNumber: Int? = null,
    val durationMinutes: Int = 30,
    val priority: Int? = null,
    val notes: String? = null,
    val appointmentDate: String? = null,
    val appointmentTime: String? = null,
    val endTime: String? = null,
    val status: String? = null,
    val remarks: String = "",
    val createdAt: String? = null,
    val updatedAt: String? = null
)
